% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{book}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  pdftitle={Introduction to Bioinformatics and Computational Biology},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage{longtable,booktabs}
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
\usepackage{booktabs}
\usepackage[]{natbib}
\bibliographystyle{plainnat}

\title{Introduction to Bioinformatics and Computational Biology}
\author{}
\date{\vspace{-2.5em}2021-04-28}

\begin{document}
\maketitle

{
\setcounter{tocdepth}{1}
\tableofcontents
}
\hypertarget{course-information}{%
\chapter{Course information}\label{course-information}}

This is the course material for STAT115/215 BIO/BST282 at Harvard University.\\
All the YouTube videos in this course are organized under the \href{https://www.youtube.com/playlist?list=PLeB-Dlq-v6taAXK6ZCGfqImrNWJzFt3p3}{2021 STAT115 playlist}.

\hypertarget{contributors}{%
\section{Contributors}\label{contributors}}

\href{http://http://liulab.dfci.harvard.edu/}{Xiaole Shirley Liu} Harvard University and Dana-Farber Cancer Institute\\
\href{https://statquest.org/}{Joshua Starmer} StatQuest\\
\href{http://wang.wustl.edu/}{Ting Wang} Washington University\\
\href{http://yuelab.org/}{Feng Yue} Northwestern University\\
\href{https://divingintogeneticsandgenomics.rbind.io/}{Ming Tang} Dana-Farber Cancer Institute\\
\href{https://www.sanger.ac.uk/group/hemberg-group/}{Martin Hemberg} Wellcome Sanger Institute\\
\href{https://www.broadinstitute.org/labs/getz}{Gad Getz} Harvard University and Broad Institute

Yang Liu\\
Bo Yuan\\
Jack Kang\\
Scarlett Qian\\
Jiazhen Rong\\
Phillip Nicol\\
Maartin De Vries

We thank many colleagues in the community, who helped Dr.~Liu in prepare the STAT115/215 BIO/BST282 course over the years. Some of the lecture slides acknowledged their contributions, but these contributors are not individually acknowledged here.

\hypertarget{intro}{%
\chapter{Introduction}\label{intro}}

\hypertarget{brief-history-of-bioinformatics}{%
\section{Brief history of bioinformatics}\label{brief-history-of-bioinformatics}}

\hypertarget{protein-structure-wave}{%
\subsection{Protein structure wave}\label{protein-structure-wave}}

\hypertarget{gene-expression-wave}{%
\subsection{Gene expression wave}\label{gene-expression-wave}}

\hypertarget{genome-sequencing-wave}{%
\subsection{Genome sequencing wave}\label{genome-sequencing-wave}}

\hypertarget{big-data-challenge-from-sequencing}{%
\subsection{Big data challenge from sequencing}\label{big-data-challenge-from-sequencing}}

\hypertarget{should-i-take-this-course}{%
\section{Should I take this course?}\label{should-i-take-this-course}}

\hypertarget{bioinformatics-vs-computational-biology}{%
\subsection{Bioinformatics vs computational biology}\label{bioinformatics-vs-computational-biology}}

\hypertarget{is-this-class-for-me}{%
\subsection{Is this class for me?}\label{is-this-class-for-me}}

\hypertarget{course-information-1}{%
\section{Course information}\label{course-information-1}}

\hypertarget{logistics}{%
\subsection{Logistics}\label{logistics}}

\hypertarget{x-shirley-liu-lab-introduction}{%
\subsection{X Shirley Liu lab introduction}\label{x-shirley-liu-lab-introduction}}

\hypertarget{lab-1}{%
\section{Lab 1}\label{lab-1}}

\hypertarget{introduction}{%
\subsection{Introduction}\label{introduction}}

\hypertarget{introduction-to-r}{%
\subsection{Introduction to R}\label{introduction-to-r}}

\hypertarget{introduction-to-bash}{%
\subsection{Introduction to Bash}\label{introduction-to-bash}}

\hypertarget{getting-started-with-cannon}{%
\subsection{Getting started with Cannon}\label{getting-started-with-cannon}}

\hypertarget{ngs}{%
\chapter{High throughput sequencing}\label{ngs}}

\hypertarget{three-generations-of-sequencing-technologies}{%
\section{Three generations of sequencing technologies}\label{three-generations-of-sequencing-technologies}}

First generation sequencing is Sanger sequencing. It is the technology that was used to obtain the first human genome sequence.

Second generation sequencing is also called next generation sequencing (NGS) and is the start of high throughput sequencing. It is what scientists use most often nowadays, and Illumina is the market leader. Most of the rest of this course will cover data analysis using second generation sequencing.

Third generation sequencing is single-molecule sequencing. There are many new technologies still under active development, although none has reached market penetration.

\hypertarget{fastq-and-fastqc}{%
\section{FASTQ and FASTQC}\label{fastq-and-fastqc}}

NGS generates FASTQ files. FASTQC is an computational approach to evaluate the quality of your NGS data.

\hypertarget{early-sequence-alignment-1-with-1}{%
\section{Early sequence alignment (1 with 1)}\label{early-sequence-alignment-1-with-1}}

In the early days (1970s), scientists were not worried about having to align too many sequences. They wanted to find the best alignment between two sequences. Many bioinformatics courses start with learning these, although it is not the main focus of our course. We included two videos in case you are interested.

The Needlemen-Wunsch algorithm is the earliest algorithm to find the alignment between two sequences and score their similarity.

When two sequences are long, and only a portion of them can align well with each other, the Smith-Waterman algorithm can find the best local sequence alignment. It is still considered the best alignment approach, although it is slow.

\hypertarget{sequence-search-algorihtms-1-with-many}{%
\section{Sequence search algorihtms (1 with many)}\label{sequence-search-algorihtms-1-with-many}}

With more and more sequences available in the public in the 1980s, scientists were interested in finding whether their newly sequenced string has been sequenced before in the public database. Therefore, the fast search algorithm BLAST was developed, using one sequence as the query to find similar sequences from a database.

\hypertarget{borrow-wheeler-aligner-many-with-many}{%
\section{Borrow-Wheeler Aligner (many with many)}\label{borrow-wheeler-aligner-many-with-many}}

With NGS, scientists need much faster search (aka mapping) algorithms in order to align the millions of sequences to the reference genome. The current best algorithm is called Borrow-Wheeler Aligner or BWA.

In order to understand BWA, we first need to introduce Borrows-Wheeler transformation and LF mapping

The basic idea of Borrows-Wheeler alignment

\hypertarget{alignment-output}{%
\section{Alignment output}\label{alignment-output}}

NGS raw data is in FASTQ. Alignment gives you SAM (alignment) or BAM (binary version of SAM) files which contain the sequence information in FASTQ and the mapping locations. BED file is the simpliest, although there is information loss.

\hypertarget{rnaseq}{%
\chapter{RNA-seq Quantification}\label{rnaseq}}

\hypertarget{introduction-to-rna-seq-experiment}{%
\section{Introduction to RNA-seq experiment}\label{introduction-to-rna-seq-experiment}}

\hypertarget{rna-quality-control-and-experimental-design}{%
\section{RNA quality control and experimental design}\label{rna-quality-control-and-experimental-design}}

\hypertarget{alignment}{%
\section{Alignment}\label{alignment}}

\hypertarget{rna-seq-qc}{%
\section{RNA-seq QC}\label{rna-seq-qc}}

\hypertarget{rna-seq-expression-index}{%
\section{RNA-seq expression index}\label{rna-seq-expression-index}}

\hypertarget{rsem-and-salmon}{%
\section{RSEM and Salmon}\label{rsem-and-salmon}}

\hypertarget{rna-seq-read-distribution}{%
\section{RNA-seq read distribution}\label{rna-seq-read-distribution}}

\hypertarget{lab-2}{%
\section{Lab 2}\label{lab-2}}

\hypertarget{star-tutorial}{%
\subsection{STAR tutorial}\label{star-tutorial}}

\hypertarget{rseqc-tutorial}{%
\subsection{RSeQC tutorial}\label{rseqc-tutorial}}

\hypertarget{rsemsalmon-tutorial}{%
\subsection{RSEM/Salmon Tutorial}\label{rsemsalmon-tutorial}}

\hypertarget{de}{%
\chapter{Differential expression, FDR, GO, and GSEA}\label{de}}

\textbf{DESeq2} is a popular and accurate computational algorithm to detect differential gene expression from RNA-seq data. It includes many elegant quantitative considerations, such as:

\begin{itemize}
\tightlist
\item
  Normalize the gene read counts by library size and composition\\
\item
  Model gene read counts with negative binomial distribution\\
\item
  Use hierarchical modeling to stabilize the gene variance\\
\item
  Use Benjamini-Hochberg to calculate control for false discovery rate of calling differentially expressed genes\\
\item
  Filter lowly expressed genes to reduce the number of hypotheses to be tested
\end{itemize}

\hypertarget{deseq2-library-normalization}{%
\section{DESeq2 library normalization}\label{deseq2-library-normalization}}

\hypertarget{deseq2-variance-stabilization}{%
\section{DESeq2 variance stabilization}\label{deseq2-variance-stabilization}}

\hypertarget{multiple-hypotheses-testing-and-false-discovery-rate}{%
\section{Multiple hypotheses testing and False Discovery Rate}\label{multiple-hypotheses-testing-and-false-discovery-rate}}

\hypertarget{deseq2-gene-filtering}{%
\section{DESeq2 gene filtering}\label{deseq2-gene-filtering}}

\hypertarget{gene-ontology-go-analysis}{%
\section{Gene Ontology (GO analysis)}\label{gene-ontology-go-analysis}}

\hypertarget{gene-set-enrichent-analysis-gsea}{%
\section{Gene Set Enrichent Analysis (GSEA)}\label{gene-set-enrichent-analysis-gsea}}

\hypertarget{deseq2-tutorial}{%
\section{DESeq2 tutorial}\label{deseq2-tutorial}}

\hypertarget{cluster}{%
\chapter{Clustering}\label{cluster}}

\hypertarget{heatmap-and-clustering-quality}{%
\section{Heatmap and clustering quality}\label{heatmap-and-clustering-quality}}

\hypertarget{hierarchical-cluster}{%
\section{Hierarchical cluster}\label{hierarchical-cluster}}

\hypertarget{k-means-cluster}{%
\section{K means cluster}\label{k-means-cluster}}

\hypertarget{pick-k-and-consensus-clustering}{%
\section{Pick K and consensus clustering}\label{pick-k-and-consensus-clustering}}

\hypertarget{batch-effect-removal}{%
\section{Batch effect removal}\label{batch-effect-removal}}

\hypertarget{lab3}{%
\section{Lab3}\label{lab3}}

\hypertarget{pca-tutorial}{%
\subsection{PCA tutorial}\label{pca-tutorial}}

\hypertarget{clustering-tutorial}{%
\subsection{Clustering tutorial}\label{clustering-tutorial}}

\hypertarget{combat-tutorial}{%
\subsection{Combat tutorial}\label{combat-tutorial}}

\hypertarget{deseq2-tutorial-1}{%
\subsection{DESeq2 Tutorial}\label{deseq2-tutorial-1}}

\hypertarget{davidgsea-tutorial}{%
\subsection{DAVID/GSEA Tutorial}\label{davidgsea-tutorial}}

\hypertarget{dr}{%
\chapter{Dimension Reduction}\label{dr}}

RNA-seq samples have tens of thousands of genes, although many genes might not vary much between samples and many others have correlated gene expression. Dimension reduction techniques aim to reduce the dimension of representing each sample with tens of thousands of genes to much fewer demensions, e.g.~2 to 100.

\hypertarget{principal-component-analysis-idea-behind-pca.}{%
\section{Principal Component Analysis: idea behind PCA.}\label{principal-component-analysis-idea-behind-pca.}}

PCA / SVD automatically outputs PC1, PC2, PC3, etc, with earlier PCs capturing the highest level of variability in the original data. Each PC is a linear combination of raw gene expression, and is orthogonal to all other PCs.

\hypertarget{principal-component-analysis-pca-applications.}{%
\section{Principal Component Analysis: PCA applications.}\label{principal-component-analysis-pca-applications.}}

PCA is a widely used method to project samples with high dimensions (e.g.~with gene expression data) onto two dimensions for better visualization. It is an intuitive way to identify sample clusters, and identify batch effect.

\hypertarget{multidimensional-scaling-mds}{%
\section{Multidimensional Scaling (MDS)}\label{multidimensional-scaling-mds}}

MDS can use differential ways to calculate pair-wise distance, then use lower dimensions to satisfy the pair-wise distance. PCA is a special case of MDS.

\hypertarget{linear-discriminant-analysis-lda}{%
\section{Linear discriminant Analysis (LDA)}\label{linear-discriminant-analysis-lda}}

LDA is not only a dimension reduction method, but also a supervised machine learning method.

\hypertarget{ml}{%
\chapter{Classification}\label{ml}}

\hypertarget{introduction-1}{%
\section{Introduction}\label{introduction-1}}

Imagine you have RNA-seq of a collection of labeled normal lung and lung cancer tissues. Given a new sample of RNA-seq from the lung with unknown diagnosis, will you be able to predict based on the existing labeled samples and the expression data whether the new sample is normal or tumor? This is a sample classification problem, and it could be solved using \textbf{unsupervised} and \textbf{supervised} learning approaches.

\textbf{Unsupervised learning} is basically clustering or dimension reduction. You can use hierarchical clustering, MDS, or PCA. After clustering and projection the data to lower dimensions, you examine the labels of the known samples (hopefully they cluster into separate groups by the label). Then you can assign label to the unknown sample based on its distance to the known samples.

\textbf{Supervised learning} considers the labels with known samples and tries to identify features that can separate the samples by the label. Cross validation is conducted to evaluate the performance of different approaches and avoid over fitting.

\href{https://statquest.org/video-index/}{StatQuest} has done an amazing job with machine learning with a full \href{https://youtube.com/playlist?list=PLblh5JKOoLUICTaGLRoHQDuF_7q2GfuJF}{playlist of well organized videos}. While the full playlist is worth a full course, for the purpose of the course, we will just highlight a number of widely used approaches. They include logistic regression (this is considered statistical machine learning), K nearest neighbors, random forest, and support vector machine (these are considered computer science machine learning).

\hypertarget{supervised-learning}{%
\section{Supervised learning}\label{supervised-learning}}

\hypertarget{cross-validation}{%
\section{Cross validation}\label{cross-validation}}

\hypertarget{regression}{%
\section{Regression}\label{regression}}

\hypertarget{regularization}{%
\section{Regularization}\label{regularization}}

\hypertarget{ridge-regression}{%
\subsection{Ridge regression}\label{ridge-regression}}

\hypertarget{lasso-regression}{%
\subsection{LASSO regression}\label{lasso-regression}}

\hypertarget{lasso-tutorial-in-r}{%
\subsubsection{LASSO tutorial in R}\label{lasso-tutorial-in-r}}

\hypertarget{knn}{%
\section{KNN}\label{knn}}

\hypertarget{decision-trees}{%
\section{Decision trees}\label{decision-trees}}

\hypertarget{random-forest}{%
\section{Random forest}\label{random-forest}}

\hypertarget{svm}{%
\section{SVM}\label{svm}}

\hypertarget{lab-4}{%
\section{Lab 4}\label{lab-4}}

\hypertarget{k-nearest-neighbors-tutorial}{%
\subsection{K-Nearest Neighbors tutorial}\label{k-nearest-neighbors-tutorial}}

\hypertarget{regressionridgelasso-tutorial}{%
\subsection{Regression/Ridge/LASSO Tutorial}\label{regressionridgelasso-tutorial}}

\hypertarget{logistic-regression-tutorial}{%
\subsection{Logistic Regression Tutorial}\label{logistic-regression-tutorial}}

\hypertarget{support-vector-machine-tutorial}{%
\subsection{Support Vector Machine Tutorial}\label{support-vector-machine-tutorial}}

\hypertarget{random-forest-tutorial}{%
\subsection{Random Forest Tutorial}\label{random-forest-tutorial}}

\hypertarget{m1re}{%
\chapter{Module I Review}\label{m1re}}

\hypertarget{module-i-review}{%
\section{Module I review}\label{module-i-review}}

\hypertarget{analysis-scenario-1}{%
\section{Analysis Scenario 1}\label{analysis-scenario-1}}

\hypertarget{analysis-scenario-2}{%
\section{Analysis Scenario 2}\label{analysis-scenario-2}}

\hypertarget{tfmf}{%
\chapter{Transcription Factor Motif Finding}\label{tfmf}}

\hypertarget{transcription-regulation}{%
\section{Transcription regulation}\label{transcription-regulation}}

\hypertarget{motif-representation}{%
\section{Motif representation}\label{motif-representation}}

\hypertarget{em}{%
\section{EM}\label{em}}

\hypertarget{gibbs-sampler}{%
\section{Gibbs sampler}\label{gibbs-sampler}}

\hypertarget{gibbs-intuition}{%
\section{Gibbs intuition}\label{gibbs-intuition}}

\hypertarget{motif-finding-in-eukaryotes}{%
\section{Motif finding in eukaryotes}\label{motif-finding-in-eukaryotes}}

\hypertarget{known-motif-database}{%
\section{Known motif database}\label{known-motif-database}}

\hypertarget{chip}{%
\chapter{ChIP-seq, Expression Integration}\label{chip}}

\hypertarget{motif-finding-in-eukaryotes-and-chip-seq}{%
\section{Motif finding in eukaryotes, and ChIP-seq}\label{motif-finding-in-eukaryotes-and-chip-seq}}

\hypertarget{macs-and-chip-seq-qc}{%
\section{MACS and ChIP-seq QC}\label{macs-and-chip-seq-qc}}

\hypertarget{identify-tf-interactions-from-chip-seq-motifs}{%
\section{Identify TF interactions from ChIP-seq motifs}\label{identify-tf-interactions-from-chip-seq-motifs}}

\hypertarget{tf-target-genes-and-expression-integration}{%
\section{TF target genes and expression integration}\label{tf-target-genes-and-expression-integration}}

\hypertarget{lab-5}{%
\section{Lab 5}\label{lab-5}}

\hypertarget{macs-tutorial}{%
\subsection{MACS Tutorial}\label{macs-tutorial}}

\hypertarget{chip-seq-qc-tutorial}{%
\subsection{ChIP-seq QC Tutorial}\label{chip-seq-qc-tutorial}}

\hypertarget{tf-motif-finding-tutorial}{%
\subsection{TF Motif Finding Tutorial}\label{tf-motif-finding-tutorial}}

\hypertarget{tf-collaborator-tutorial}{%
\subsection{TF Collaborator Tutorial}\label{tf-collaborator-tutorial}}

\hypertarget{epi}{%
\chapter{Epigenetics, DNA Methylation}\label{epi}}

\hypertarget{intro-to-dna-methylation}{%
\section{Intro to DNA Methylation}\label{intro-to-dna-methylation}}

\hypertarget{dna-methylation-pattern-and-function}{%
\section{DNA Methylation Pattern and Function}\label{dna-methylation-pattern-and-function}}

\hypertarget{dna-methylation-in-diseases}{%
\section{DNA Methylation in Diseases}\label{dna-methylation-in-diseases}}

\hypertarget{techniques-to-measure-dna-methylation}{%
\section{Techniques to Measure DNA Methylation}\label{techniques-to-measure-dna-methylation}}

\hypertarget{hist}{%
\chapter{Histone Modifications , Chromatin Accessibility}\label{hist}}

\hypertarget{nucleosome-positioning}{%
\section{Nucleosome Positioning}\label{nucleosome-positioning}}

\hypertarget{introduction-to-histone-modifications}{%
\section{Introduction to Histone Modifications}\label{introduction-to-histone-modifications}}

\hypertarget{infer-transcription-factor-binding-from-histone-mark-dynamics}{%
\section{Infer Transcription Factor Binding from Histone Mark Dynamics}\label{infer-transcription-factor-binding-from-histone-mark-dynamics}}

\hypertarget{using-histone-marks-to-infer-gene-functions}{%
\section{Using Histone Marks to Infer Gene Functions}\label{using-histone-marks-to-infer-gene-functions}}

\hypertarget{introduction-to-dnase-seq-and-atac-seq}{%
\section{Introduction to DNase-seq and ATAC-seq}\label{introduction-to-dnase-seq-and-atac-seq}}

\hypertarget{infer-tf-from-differential-genes-using-lisa}{%
\section{Infer TF from Differential Genes Using LISA}\label{infer-tf-from-differential-genes-using-lisa}}

\hypertarget{caution-on-dnaseatac-seq-footprint-analysis}{%
\section{Caution on DNase/ATAC-seq footprint analysis}\label{caution-on-dnaseatac-seq-footprint-analysis}}

\hypertarget{summary-of-epigenetics-and-chromatin}{%
\section{Summary of Epigenetics and Chromatin}\label{summary-of-epigenetics-and-chromatin}}

\hypertarget{lab-6}{%
\section{Lab 6}\label{lab-6}}

\hypertarget{chip-seq-expression-integration}{%
\subsection{ChIP-seq Expression Integration}\label{chip-seq-expression-integration}}

\hypertarget{cistrome-go-tutorial}{%
\subsection{Cistrome-GO Tutorial}\label{cistrome-go-tutorial}}

\hypertarget{atac-seq-analysis-and-lisa-tutorial}{%
\subsection{ATAC-seq Analysis and LISA Tutorial}\label{atac-seq-analysis-and-lisa-tutorial}}

\hypertarget{hmm}{%
\chapter{Hidden Markov Model}\label{hmm}}

\hypertarget{markov-chain}{%
\section{Markov Chain}\label{markov-chain}}

\hypertarget{hidden-markov-model}{%
\section{Hidden Markov Model}\label{hidden-markov-model}}

\hypertarget{hidden-markov-model-forward-procedure}{%
\section{Hidden Markov Model Forward Procedure}\label{hidden-markov-model-forward-procedure}}

\hypertarget{hidden-markov-model-backward-procedure}{%
\section{Hidden Markov Model Backward Procedure}\label{hidden-markov-model-backward-procedure}}

\hypertarget{hmm-forward-backward-algorithm}{%
\section{HMM Forward-Backward Algorithm}\label{hmm-forward-backward-algorithm}}

\hypertarget{viterbi-algorithm}{%
\section{Viterbi Algorithm}\label{viterbi-algorithm}}

\hypertarget{baum-welch-algorithm-intuition}{%
\section{Baum Welch Algorithm Intuition}\label{baum-welch-algorithm-intuition}}

\hypertarget{hmm-bioinformatics-applications}{%
\section{HMM Bioinformatics Applications}\label{hmm-bioinformatics-applications}}

\hypertarget{HiC}{%
\chapter{HiC}\label{HiC}}

\hypertarget{introduction-to-chromatin-interaction-and-organization}{%
\section{Introduction to Chromatin Interaction and Organization}\label{introduction-to-chromatin-interaction-and-organization}}

\hypertarget{methods-to-investigate-3d-genome-organization}{%
\section{Methods to Investigate 3D Genome Organization}\label{methods-to-investigate-3d-genome-organization}}

\hypertarget{topologically-associating-domains}{%
\section{Topologically Associating Domains}\label{topologically-associating-domains}}

\hypertarget{tad-function-and-loop-anchors}{%
\section{TAD Function and Loop Anchors}\label{tad-function-and-loop-anchors}}

\hypertarget{chromatin-compartments}{%
\section{Chromatin Compartments}\label{chromatin-compartments}}

\hypertarget{computational-methods-to-call-chromatin-loops}{%
\section{Computational Methods to Call Chromatin Loops}\label{computational-methods-to-call-chromatin-loops}}

\hypertarget{variations-of-chromatin-interaction-technologies}{%
\section{Variations of Chromatin Interaction Technologies}\label{variations-of-chromatin-interaction-technologies}}

\hypertarget{resources-for-exploring-3d-genomes}{%
\section{Resources for Exploring 3D Genomes}\label{resources-for-exploring-3d-genomes}}

\hypertarget{lab-7}{%
\section{Lab 7}\label{lab-7}}

\hypertarget{bs-seq-and-bismark-tutorial}{%
\subsection{BS-seq and Bismark Tutorial}\label{bs-seq-and-bismark-tutorial}}

\hypertarget{tutorial-on-associating-dna-methylation-with-expression}{%
\subsection{Tutorial on Associating DNA Methylation with Expression}\label{tutorial-on-associating-dna-methylation-with-expression}}

\hypertarget{hic-analysis-tutorial}{%
\subsection{HiC Analysis Tutorial}\label{hic-analysis-tutorial}}

\hypertarget{m2re}{%
\chapter{Module II Review}\label{m2re}}

\hypertarget{module-ii-review}{%
\section{Module II Review}\label{module-ii-review}}

\hypertarget{module-ii-analysis-scenarios}{%
\section{Module II Analysis Scenarios}\label{module-ii-analysis-scenarios}}

\hypertarget{gwas1}{%
\chapter{SNP and GWAS}\label{gwas1}}

\hypertarget{snp-lp-and-association-studies}{%
\section{SNP, LP, and Association Studies}\label{snp-lp-and-association-studies}}

\hypertarget{gwas-studies-and-eqtl-analysis}{%
\section{GWAS Studies and eQTL Analysis}\label{gwas-studies-and-eqtl-analysis}}

\hypertarget{lab-8}{%
\section{Lab 8}\label{lab-8}}

\hypertarget{hw4-faq-cooler}{%
\subsection{HW4 FAQ \& cooler}\label{hw4-faq-cooler}}

\hypertarget{pikachuhiglass}{%
\subsection{Pikachu\&HiGlass}\label{pikachuhiglass}}

\hypertarget{hmm-1}{%
\subsection{HMM}\label{hmm-1}}

\hypertarget{gwas2}{%
\chapter{GWAS and Epigenomics}\label{gwas2}}

\hypertarget{intro-functional-annotate-gwas}{%
\section{Intro Functional Annotate GWAS}\label{intro-functional-annotate-gwas}}

\hypertarget{gwas-functional-enrichment}{%
\section{GWAS Functional Enrichment}\label{gwas-functional-enrichment}}

\hypertarget{find-causal-snps}{%
\section{Find Causal SNPs}\label{find-causal-snps}}

\hypertarget{predict-disease-risk}{%
\section{Predict disease risk}\label{predict-disease-risk}}

\hypertarget{scrna1}{%
\chapter{Single-cell RNA-seq (1)}\label{scrna1}}

\hypertarget{intro-to-scrna-seq}{%
\section{Intro to scRNA-seq}\label{intro-to-scrna-seq}}

\hypertarget{scrna-seq-techniques}{%
\section{scRNA seq techniques}\label{scrna-seq-techniques}}

\hypertarget{scrna-seq-preprocessing-and-qc}{%
\section{scRNA seq preprocessing and QC}\label{scrna-seq-preprocessing-and-qc}}

\hypertarget{cleaning-up-expression-matrix}{%
\section{Cleaning up expression matrix}\label{cleaning-up-expression-matrix}}

\hypertarget{scrna2}{%
\chapter{Single-cell RNA-seq (2)}\label{scrna2}}

\hypertarget{scrna-seq-dimension-reduction}{%
\section{scRNA seq dimension reduction}\label{scrna-seq-dimension-reduction}}

\hypertarget{clustering-and-projections}{%
\section{Clustering and projections}\label{clustering-and-projections}}

\hypertarget{pseudo-time-and-rna-velocity}{%
\section{Pseudo time and RNA velocity}\label{pseudo-time-and-rna-velocity}}

\hypertarget{clustering-by-genotype-and-cite-seq}{%
\section{Clustering by genotype and CITE seq}\label{clustering-by-genotype-and-cite-seq}}

\hypertarget{scatac}{%
\chapter{scATAC-seq}\label{scatac}}

\hypertarget{single-cell-atac-seq-technique}{%
\section{Single-Cell ATAC-seq Technique}\label{single-cell-atac-seq-technique}}

\hypertarget{single-cell-atac-seq-pre-processing-and-qc}{%
\section{Single-Cell ATAC-seq Pre-Processing and QC}\label{single-cell-atac-seq-pre-processing-and-qc}}

\hypertarget{single-cell-atac-seq-analysis}{%
\section{Single-Cell ATAC-seq Analysis}\label{single-cell-atac-seq-analysis}}

\hypertarget{scatac-seq-downstream-analyses-and-scrna-seq-integration}{%
\section{scATAC-seq Downstream Analyses and scRNA-seq Integration}\label{scatac-seq-downstream-analyses-and-scrna-seq-integration}}

\hypertarget{lab-9}{%
\section{Lab 9}\label{lab-9}}

\hypertarget{maestro-tutorial}{%
\subsection{MAESTRO tutorial}\label{maestro-tutorial}}

\hypertarget{m3re}{%
\chapter{Module III Review}\label{m3re}}

\hypertarget{module-iii-review}{%
\section{Module III Review}\label{module-iii-review}}

\hypertarget{cancerseq}{%
\chapter{Cancer Genome Sequencing , Mutation analyses}\label{cancerseq}}

\hypertarget{intro-to-tcga}{%
\section{Intro to TCGA}\label{intro-to-tcga}}

\hypertarget{cancer-mutation-characterization}{%
\section{Cancer mutation characterization}\label{cancer-mutation-characterization}}

\hypertarget{cancer-mutation-patterns}{%
\section{Cancer mutation patterns}\label{cancer-mutation-patterns}}

\hypertarget{tumor-purity-and-clonality}{%
\section{Tumor purity and clonality}\label{tumor-purity-and-clonality}}

\hypertarget{interpret-tumor-mutations}{%
\section{Interpret tumor mutations}\label{interpret-tumor-mutations}}

\hypertarget{find-cancer-genes}{%
\section{Find cancer genes}\label{find-cancer-genes}}

\hypertarget{summary-and-future}{%
\section{Summary and future}\label{summary-and-future}}

\hypertarget{cancersub}{%
\chapter{Cancer Subtyping, Survival Analyses}\label{cancersub}}

\hypertarget{tumor-subtypes}{%
\section{Tumor Subtypes}\label{tumor-subtypes}}

\hypertarget{survival-analysis}{%
\section{Survival analysis}\label{survival-analysis}}

\hypertarget{oncogenes-and-tumor-suppressor-mutations}{%
\section{Oncogenes and Tumor Suppressor Mutations}\label{oncogenes-and-tumor-suppressor-mutations}}

\hypertarget{cancer-epigenetics}{%
\section{Cancer Epigenetics}\label{cancer-epigenetics}}

\hypertarget{PrecisionMedicine}{%
\chapter{Targeted Therapy and Precision Medicine}\label{PrecisionMedicine}}

\hypertarget{introduction-to-targeted-therapy-and-precision-medicine}{%
\section{Introduction to Targeted Therapy and Precision Medicine}\label{introduction-to-targeted-therapy-and-precision-medicine}}

\hypertarget{resistance-to-targeted-therapy}{%
\section{Resistance to targeted therapy}\label{resistance-to-targeted-therapy}}

\hypertarget{model-system-chemical-and-genetic-screens}{%
\section{Model system chemical and genetic screens}\label{model-system-chemical-and-genetic-screens}}

\hypertarget{overcoming-resistance-to-targeted-therapy}{%
\section{Overcoming resistance to targeted therapy}\label{overcoming-resistance-to-targeted-therapy}}

\hypertarget{cancerimmuno1}{%
\chapter{Cancer Immunotherapy (1)}\label{cancerimmuno1}}

\hypertarget{intro-to-cancer-immunotherapy}{%
\section{Intro to Cancer Immunotherapy}\label{intro-to-cancer-immunotherapy}}

\hypertarget{hla-and-neoantigen-presentation}{%
\section{HLA and Neoantigen Presentation}\label{hla-and-neoantigen-presentation}}

\hypertarget{immune-cell-infiltration-in-tumors}{%
\section{Immune Cell Infiltration in Tumors}\label{immune-cell-infiltration-in-tumors}}

\hypertarget{t-cell-receptor-repertoires-in-cancer-immunology}{%
\section{T Cell Receptor Repertoires in Cancer Immunology}\label{t-cell-receptor-repertoires-in-cancer-immunology}}

\hypertarget{lab-10}{%
\section{Lab 10}\label{lab-10}}

\hypertarget{tcga-exploration}{%
\subsection{TCGA exploration}\label{tcga-exploration}}

\hypertarget{limma-on-microarray-data}{%
\subsection{LIMMA on microarray data}\label{limma-on-microarray-data}}

\hypertarget{survival-analysis-1}{%
\subsection{Survival analysis}\label{survival-analysis-1}}

\hypertarget{cancerimmuno2}{%
\chapter{Cancer Immunotherapy (2)}\label{cancerimmuno2}}

\hypertarget{b-cell-receptor-repertoires-in-tumors}{%
\section{B Cell Receptor Repertoires in Tumors}\label{b-cell-receptor-repertoires-in-tumors}}

\hypertarget{t-cell-activation-and-dysfunction}{%
\section{T Cell Activation and Dysfunction}\label{t-cell-activation-and-dysfunction}}

\hypertarget{nk-cells-and-macrophages-in-tumor-immunity}{%
\section{NK Cells and Macrophages in Tumor Immunity}\label{nk-cells-and-macrophages-in-tumor-immunity}}

\hypertarget{cancer-immunotherapy-response-biomarkers}{%
\section{Cancer Immunotherapy Response Biomarkers}\label{cancer-immunotherapy-response-biomarkers}}

\hypertarget{improving-immunotherapy-response}{%
\section{Improving Immunotherapy Response}\label{improving-immunotherapy-response}}

\hypertarget{lab-11}{%
\section{Lab 11}\label{lab-11}}

\hypertarget{cancer-mutations-and-driver-genes}{%
\subsection{Cancer mutations and driver genes}\label{cancer-mutations-and-driver-genes}}

\hypertarget{cripsr-screen}{%
\subsection{CRIPSR screen}\label{cripsr-screen}}

\hypertarget{cancer-immunology}{%
\subsection{Cancer immunology}\label{cancer-immunology}}

\hypertarget{crispr}{%
\chapter{CRISPR Screens}\label{crispr}}

\hypertarget{introduction-to-crispr-and-crispr-screens}{%
\section{Introduction to CRISPR and CRISPR Screens}\label{introduction-to-crispr-and-crispr-screens}}

\hypertarget{computational-resources-for-crispr-and-screens}{%
\section{Computational Resources for CRISPR and Screens}\label{computational-resources-for-crispr-and-screens}}

\hypertarget{cancer-cell-vulnerability-from-crispr-screens}{%
\section{Cancer Cell Vulnerability from CRISPR Screens}\label{cancer-cell-vulnerability-from-crispr-screens}}

\hypertarget{immune-related-crispr-screens}{%
\section{Immune Related CRISPR Screens}\label{immune-related-crispr-screens}}

\hypertarget{depmap-tutorial}{%
\section{Depmap tutorial}\label{depmap-tutorial}}

\hypertarget{m4re}{%
\chapter{Module IV Review and Course Review}\label{m4re}}

\hypertarget{module-iv-review}{%
\section{Module IV Review}\label{module-iv-review}}

\hypertarget{final-course-review}{%
\section{Final Course Review}\label{final-course-review}}

\hypertarget{final-exam-preparations}{%
\section{Final Exam Preparations}\label{final-exam-preparations}}

\hypertarget{levels-of-bioinformatics-and-preparing-for-the-future}{%
\section{Levels of Bioinformatics and Preparing for the Future}\label{levels-of-bioinformatics-and-preparing-for-the-future}}

\end{document}
